### PR TITLE
Add Gateio exchange

### PR DESCRIPTION
**Pull request by aliyagu**
### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [ ] Did you add new tests?
- [ ] Did you add reviewers?

Add Gateio exchange

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -43,6 +43,13 @@ CryptoAPIs is a library written in Julia that combines API wrappers from various
                 <td><a href="src/Coinbase/Spot">CryptoAPIs.Coinbase.Spot</a></td>
                 <td><a href="https://bhftbootcamp.github.io/CryptoAPIs.jl/stable/pages/Coinbase/#Spot">Spot</a></td>
             </tr>
+            <tr>
+                <td><img src="docs/src/assets/gateio.png" alt="Gateio Logo" width="20" height="20"></td>
```
Comment (artememelin) [inline docs/src/index.md:47]:
```suggestion
                <td><img src="assets/gateio.png" alt="Gateio Logo" width="20" height="20"></td>
```

File: src/Gateio/Spot/API/Candle.jl [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    low_price::Maybe{Float64}
+    open_price::Maybe{Float64}
+    base_amount::Maybe{Float64}
+end
+
+"""
+    candle(client::GateioClient, query::CandleQuery)
+    candle(client::GateioClient = Gateio.Spot.public_client; kw...)
+
+Market candlesticks
```
Comment (artememelin) [inline src/Gateio/Spot/API/Candle.jl:51]:
```suggestion
Market candlesticks.
```

File: src/Gateio/Spot/API/Deposit.jl [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+function Serde.isempty(::Type{<:DepositData}, x)::Bool
+    return x === ""
+end
+
+"""
+    deposit(client::GateioClient, query::CandleQuery)
+    deposit(client::GateioClient; kw...)
+
+Retrieve deposit records
```
Comment (artememelin) [inline src/Gateio/Spot/API/Deposit.jl:48]:
```suggestion
Retrieve deposit records.
```

File: src/Gateio/Gateio.jl [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    return sign
+end
+
+function CryptoAPIs.request_sign!(client::GateioClient, query::Q, endpoint::String)::Q where {Q<:GateioPrivateQuery}
+    query.signTimestamp = Dates.now(UTC)
+    query.sign = nothing
+    endpoint = "/" * endpoint
+    query.sign = hexdigest("sha512", client.secret_key, gen_sign("GET", query, endpoint))
+    return query
+end
```
Comment (dmitrii-doronin) [inline src/Gateio/Gateio.jl:91]: Maybe we could get rid of `gen_sign` if it's the place it is used?

File: src/Gateio/Gateio.jl [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+function CryptoAPIs.request_headers(client::GateioClient, ::GateioPublicQuery)::Vector{Pair{String,String}}
+    return Pair{String,String}[
+        "Content-Type" => "application/json"
+    ]
+end
+
+function CryptoAPIs.request_headers(client::GateioClient, query::GateioPrivateQuery)::Vector{Pair{String,String}}
+    return Pair{String,String}[
+        "KEY"          => client.public_key,
+        "SIGN"         => query.sign,
```
Comment (dmitrii-doronin) [inline src/Gateio/Gateio.jl:110]: `query.signature` would be more in terms with other packages

File: src/Gateio/Utils.jl [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    return unixnanos2nanodate(parse(Int64, x) * 1e9)
+end
+
+function Serde.SerQuery.ser_value(::Type{<:GateioCommonQuery}, ::Val{:signTimestamp}, v::Any)::Nothing
+    return nothing
+end
+
+function Serde.SerQuery.ser_value(::Type{<:GateioCommonQuery}, ::Val{:sign}, v::Any)::Nothing
+    return nothing
+end
```
Comment (dmitrii-doronin) [inline src/Gateio/Utils.jl:17]: There's a  `ignore_field(::Type{T}, ::Val{x})::Bool where {T,x} = false` method in Serde defined for specifically for that case.

File: src/Gateio/Gateio.jl (hunk 5 above)
Comment (gryumov) [reply src/Gateio/Gateio.jl:110]: you can't, it's converted to a request

File: src/Gateio/Utils.jl (hunk 6 above)
Comment (gryumov) [reply src/Gateio/Utils.jl:17]: `ser_ignore_field`

**Review by gryumov — APPROVED** [review]

**Pull request merged**